library/Reporting/Report.php

```diff
@@ -30,98 +30,39 @@ class Report
     /** @var Reportlet[] */
     protected $reportlets;
 
-    /** @var Schedule */
-    protected $schedule;
-
     /** @var Template */
     protected $template;
 
+    /** @var Schedule */
+    protected $schedule;
+
     /**
-     * @param int $id
-     *
-     * @return  static
+     * @param Model\Report $reportModel
      *
-     * @throws  Exception
+     * @return Report
      */
-    public static function fromDb($id)
+    public static function fromModel(Model\Report $reportModel): Report
     {
         $report = new static();
 
-        $db = $report->getDb();
-
-        $select = (new Sql\Select())
-            ->from('report')
-            ->columns('*')
-            ->where(['id = ?' => $id]);
+        $report->id = $reportModel->id;
+        $report->name = $reportModel->name;
+        $report->author = $reportModel->author;
+        $report->timeframe = Timeframe::fromModel($reportModel->timeframe);
+        $report->template = Template::fromModel($reportModel->template);
 
-        $row = $db->select($select)->fetch();
-
-        if ($row === false) {
-            throw new Exception('Report not found');
-        }
-
-        $report
-            ->setId($row->id)
-            ->setName($row->name)
-            ->setAuthor($row->author)
-            ->setTimeframe(Timeframe::fromDb($row->timeframe_id))
-            ->setTemplate(Template::fromDb($row->template_id));
-
-        $select = (new Sql\Select())
-            ->from('reportlet')
-            ->columns('*')
-            ->where(['report_id = ?' => $id]);
-
-        $row = $db->select($select)->fetch();
-
-        if ($row === false) {
+        if ($reportModel->reportlet === null) {
             throw new Exception('No reportlets configured.');
         }
 
-        $reportlet = new Reportlet();
-
-        $reportlet
-            ->setId($row->id)
-            ->setClass($row->class);
+        $reportModel->reportlet->report_name = $reportModel->name;
+        $reportModel->reportlet->report_id = $reportModel->id;
+        $reportlet = Reportlet::fromModel($reportModel->reportlet);
 
-        $select = (new Sql\Select())
-            ->from('config')
-            ->columns('*')
-            ->where(['reportlet_id = ?' => $row->id]);
+        $report->reportlets = [$reportlet];
 
```

Comment:
When a report is supposed to contain more than one reportlet, you should also name the relation accordingly. I.e. the name of the relation should be `reportlets` instead of `reportlet` and you can proceed as follows to transform it from the model to the `Reportlet` class.

```php
$reportlets = [];
foreach($reportModel->reportlets as $reportlet) {
    $reportlets[] = Reportlet::fromModel($reportlet);
}

$report->reportlets = $reportlets;
```